src/oas/resolver.ts

```diff
@@ -23,8 +23,18 @@ export function setSharedKey(value: unknown, key: string) {
   return false;
 }
 
-export function getComponentName(references: References, $ref: string) {
-  return getResolved(references, $ref).match(/^#\/components\/([A-Za-z]+)\//)?.[1];
+const COMPONENTS_FRAGMENT_PATTERN = /#\/components\/(?<section>[A-Za-z0-9_-]+)\//;
+
+/**
+ * Given a URI, return the name of the components sub-section indicated.  For
+ * example, given `#/components/securitySchemes` returns "securitySchemes".
+ * @returns the name of the components sub-section, or undefined if none is
+ * present
+ */
+export function getComponentName(references: References, $ref: string): string | undefined {
+  const uri = getResolved(references, $ref);
+  const match = uri.match(COMPONENTS_FRAGMENT_PATTERN);
+  return match?.groups?.['section'];
```

Comment:
The previous expression didn't allow for `x-` extension properties in the `components` section.